testing/endtoend/evaluators/beaconapi_evaluators/beacon_api.go

```diff
@@ -0,0 +1,440 @@
+			castedp, ok := prysmResp.(*apimiddleware.IdentityResponseJson)
+			if !ok {
+				return errors.New("failed to cast type")
+			}
+			castedl, ok := lhouseResp.(*apimiddleware.IdentityResponseJson)
+			if !ok {
+				return errors.New("failed to cast type")
+			}
+			if castedp.Data == nil {
+				return errors.New("prysm node identity was empty")
+			}
+			if castedl.Data == nil {
+				return errors.New("lighthouse node identity was empty")
+			}
+			return nil
+		},
+	},
+	"/node/peers": {
+		basepath: v1MiddlewarePathTemplate,
+		params: func(_ string, _ types.Epoch) []string {
+			return []string{"skip"}
+		},
+		prysmResps: map[string]interface{}{
+			"json": &apimiddleware.PeersResponseJson{},
+		},
+		lighthouseResps: map[string]interface{}{
+			"json": &apimiddleware.PeersResponseJson{},
+		},
+		customEvaluation: func(prysmResp interface{}, lhouseResp interface{}) error {
+			castedp, ok := prysmResp.(*apimiddleware.PeersResponseJson)
+			if !ok {
+				return errors.New("failed to cast type")
+			}
+			castedl, ok := lhouseResp.(*apimiddleware.PeersResponseJson)
+			if !ok {
+				return errors.New("failed to cast type")
+			}
+			if castedp.Data == nil {
+				return errors.New("prysm node identity was empty")
+			}
+			if castedl.Data == nil {
+				return errors.New("lighthouse node identity was empty")
+			}
+			return nil
+		},
+	},
+}
+
+func withCompareBeaconAPIs(beaconNodeIdx int, conn *grpc.ClientConn) error {
+	ctx := context.Background()
+	beaconClient := service.NewBeaconChainClient(conn)
+	genesisData, err := beaconClient.GetGenesis(ctx, &empty.Empty{})
+	if err != nil {
+		return errors.Wrap(err, "error getting genesis data")
+	}
+	currentEpoch := slots.EpochsSinceGenesis(genesisData.Data.GenesisTime.AsTime())
+
+	for path, meta := range beaconPathsAndObjects {
+		for key := range meta.prysmResps {
+			switch key {
+			case "json":
+				jsonparams := meta.params("json", currentEpoch)
+				if len(jsonparams) == 0 {
+					continue
+				}
+				apipath := pathFromParams(path, jsonparams)
+				fmt.Printf("json api path: %s", apipath)
+				if err := compareJSONMulticlient(beaconNodeIdx,
+					meta.basepath,
+					apipath,
+					beaconPathsAndObjects[path].prysmResps[key],
+					beaconPathsAndObjects[path].lighthouseResps[key],
+					meta.customEvaluation,
+				); err != nil {
+					return err
+				}
+				fmt.Printf("prysm ob: %v/n", beaconPathsAndObjects[path].prysmResps[key])
+				fmt.Printf("lighthouse ob: %v", beaconPathsAndObjects[path].prysmResps[key])
+			case "ssz":
+				sszparams := meta.params("ssz", currentEpoch)
+				if len(sszparams) == 0 {
+					continue
+				}
+				apipath := pathFromParams(path, sszparams)
+				fmt.Printf("ssz api path: %s", apipath)
+				prysmr, lighthouser, err := compareSSZMulticlient(beaconNodeIdx, meta.basepath, apipath)
+				if err != nil {
+					return err
+				}
+				beaconPathsAndObjects[path].prysmResps[key] = prysmr
+				beaconPathsAndObjects[path].lighthouseResps[key] = lighthouser
```

Comment:
Is this needed?

Reply:
yeah it's later used in the unmarshalling and comparing the unmarshals